qemu/target-i386/cpu.h

```diff
@@ -699,7 +699,7 @@ typedef enum {
 
 typedef struct SegmentCache {
     uint32_t selector;
-    target_ulong base;
+    uint64_t base;   /* handle 32 or 64 bit CPUs */
```

Comment:
no, target_long is 32bit for 32bit machine, and 64bit for 64bit machine. this is how Qemu handles long data type for different kind of guests.

we can't cast SegmentCache to our struct, but we can just copy each field like what you are doing now, can't we?